src/wikitextprocessor/lua/_sandbox_phase1.lua

```diff
@@ -12,20 +14,55 @@ local loader_cache = {}
 local loaddata_cache = {}
 local _orig_package = package
 
+-- Copied from https://github.com/wikimedia/mediawiki-extensions-Scribunto/blob/8d69dc173e33ae936ff4401d41ee5e6a1fd1ba67/includes/Engines/LuaCommon/lualib/mwInit.lua#L38-L66
+--- Do a "deep copy" of a table or other value.
+function mw_clone( val )
+    local tableRefs = {}
+    local function recursiveClone( val )
+        if type( val ) == 'table' then
+            -- Encode circular references correctly
+            if tableRefs[val] ~= nil then
+                return tableRefs[val]
+            end
+
+            local retVal
+            retVal = {}
+            tableRefs[val] = retVal
+
+            -- Copy metatable
+            if getmetatable( val ) then
+                setmetatable( retVal, recursiveClone( getmetatable( val ) ) )
+            end
+
+            for key, elt in pairs( val ) do
+                retVal[key] = recursiveClone( elt )
+            end
+            return retVal
+        else
+            return val
+        end
+    end
+    return recursiveClone( val )
+end
+
 -- This function loads new a new module, whether built-in or defined in the
 -- data file, and returns its initialization function.  This caches the
 -- initialization function.
-function new_loader(modname)
-   -- this module wreaks havoc on the global namespace
-   if modname == "Module:No globals" then
```

Comment:
This Lua module is removed from Wiktionary and this code was added because Lua environments are not isolated(https://github.com/tatuylonen/wikitextprocessor/commit/faa4bfce87cd7cfc37f0a2b5d39460892d0c48ce).